sql/api/src/main/scala/org/apache/spark/sql/errors/ExecutionErrors.scala

```diff
@@ -195,7 +195,7 @@ private[sql] trait ExecutionErrors extends DataTypeErrorsBase {
       fieldName: String,
       walkedTypePath: WalkedTypePath): SparkUnsupportedOperationException = {
     new SparkUnsupportedOperationException(
-      errorClass = "_LEGACY_ERROR_TEMP_2140",
+      errorClass = "INVALID_JAVA_IDENTIFIER_AS_FIELD_NAME",
       messageParameters =
         Map("fieldName" -> fieldName, "walkedTypePath" -> walkedTypePath.toString))
```

Comment:
I think we can consider `fieldName` as an id, and quote it by `toSQLId`

Reply:
Sounds good. Just updated, thanks!